captum/attr/_core/lime.py

```diff
@@ -481,7 +501,7 @@ def attribute(
             dataset = TensorDataset(
                 combined_interp_inps, combined_outputs, combined_sim
             )
-            self.interpretable_model.fit(DataLoader(dataset, batch_size=n_samples))
+            self.interpretable_model.fit(DataLoader(dataset, batch_size=batch_count))
```

Comment:
nit: Wouldn't `n_samples` still work here even in case of `perturb_generator` ?

Reply:
The edge case is the same as above, essentially a provided generator which provides fewer than n_samples. A warning in this case (rather than an error) requires this change.